### PR TITLE
Avoid returning biggest resolution when fpr > 0.283

### LINKED ISSUE #57032: Unexpected behaviour for bloomfilter secondary index with false positive > 0.283
Bloom filter secondary index accepts false positive rate as a parameter (0.025 by default).

This is the function where bloom filter configuration parameters are chosen based on the provided false positive rate:

https://github.com/ClickHouse/ClickHouse/blob/4ed4fbe74df0f48382bad2c55c883b763e5c869f/src/Interpreters/BloomFilterHash.h#L246-L291

Since there are no comments explaining how the `probability_lookup_table` was generated is a bit difficult to guess how that table is supposed to be used.

First thing, why `min_probability_index_each_bits[] = {0, 0, 1, 2, 3, 3, 4, 5, 6, 6, 7, 8, 8, 9, 10, 10, 11, 12, 12, 13, 14}` and not `min_probability_index_each_bits[] = {0, 1, 2, 3, 4, 5, 6, 6, 7, 8, 8, 8, 8, 9, 10, 10, 11, 12, 12, 13, 14}`? With the former, there are values in the lookup that are not being used.

Also, users are allowed to provide any false positive rate between 0 and 1 but for values bigger than 0.283 the function always returns `(19, 13)` which is the maximum resolution allowed. 

We should either allow only values between 0 and 0.283 or try to fix the output when the provided false positive rate is too big.

cc @zhang2014

### PR DESCRIPTION
**Pull request by jrdi**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Closes https://github.com/ClickHouse/ClickHouse/issues/57032

### CODE REVIEW

File: src/Interpreters/BloomFilterHash.h [hunk 1]
```diff
@@ -248,6 +248,13 @@ struct BloomFilterHash
         static const size_t MAX_BITS_PER_ROW = 20;
         static const size_t MAX_HASH_FUNCTION_COUNT = 15;
 
+        static const size_t MIN_BITS_PER_ROW = 2;
+        static const size_t MIN_HASH_FUNCTION_COUNT = 2;
+
+        /// Return the smaller possible parameters for false positive rates bigger than 0.283
```
Comment (alexey-milovidov) [inline src/Interpreters/BloomFilterHash.h:254]: Need the reasoning behind 0.283.

**Comment by alexey-milovidov** [comment]
Needs a test.

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit fd792a80230c954393bd4720cf46e041d535deef with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57034/fd792a80230c954393bd4720cf46e041d535deef/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6942147955">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57034/fd792a80230c954393bd4720cf46e041d535deef/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Interpreters/BloomFilterHash.h (hunk 1 above)
Comment (jrdi) [reply src/Interpreters/BloomFilterHash.h:254]: Ok, I'll improve the comment

**Comment by jrdi** [comment]
> Needs a test.

I guess the only way to test this function properly is by writing a unit test using `gtest`. I'll give it a try.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**